### PR TITLE
Update docs for new developers

### PR DESCRIPTION
**Pull request by aiwilliams**
Based on working to get @ndowmon going on development.

### CODE REVIEW

File: docs/development.md [hunk 1]
```diff
@@ -62,21 +76,74 @@ Creating a role assignment under the scope of "/subscriptions/dccea..."
...
+```txt
+CLIENT_ID=app_registration_client_id
+CLIENT_SECRET=app_registration_client_secret
+DIRECTORY_ID=development_directory_id
+SUBSCRIPTION_ID=development_subscription_id
+INGEST_ACTIVE_DIRECTORY=true
 ```
 
-## Terraforming for Development
+You should not be able to [execute the integration](#execute-the-integration)
```
Comment (ndowmon) [inline docs/development.md:130]: Is this supposed to say "You should _now_ be able to execute the integration"?

File: docs/development.md [hunk 2]
```diff
@@ -62,21 +76,74 @@ Creating a role assignment under the scope of "/subscriptions/dccea..."
...
+with these credentials. Note that [running the tests](#running-tests) requires
+more attention.
+
+## Executing Terraform
+
+Example Azure resources are maintained with Terraform to allow for easy setup
+and teardown, and to avoid unnecessary expenses between development cycles. See
+Terraform code where `count` is used to force a decision about which resources
+to build. Once an integration step has a recording made, those resources can be
+set back to count `0`.
```
Comment (ndowmon) [inline docs/development.md:140]: For my own info, when you do want to generate some of these resources, are you simply resetting the `default` values in `terraform/vars.tf`? Or do you override them with environment variables?

File: docs/development.md [hunk 3]
```diff
@@ -62,21 +76,74 @@ Creating a role assignment under the scope of "/subscriptions/dccea..."
...
+Using the Application (client) ID, Directory (tenant) ID, and generated secret
+from the App Registration, create `<graph-azure>/.env` to establish
+authentication values for the integration API clients. This file is already in
+`.gitignore`, and is distinct from the `<graph-azure>/terraform/.env` file read
+by Terraform.
+
+```txt
+CLIENT_ID=app_registration_client_id
+CLIENT_SECRET=app_registration_client_secret
+DIRECTORY_ID=development_directory_id
```
Comment (ndowmon) [inline docs/development.md:125]: not sure if it's good to postpend `# same as above` to DIRECTORY_ID and SUBSCRIPTION_ID or `#different from above` to CLIENT_ID and CLIENT_SECRET, these two sets of vars in the same document do tend to look similar and could maybe be confused without a specific callout?

File: docs/development.md (hunk 2 above)
Comment (aiwilliams) [reply docs/development.md]: Add a tip, that's a great way to do it!

**Pull request merged**